torch/distributed/pipelining/schedules.py

```diff
@@ -1123,10 +1124,14 @@ def __init__(
         self.pp_group_size = stages[0].group_size
         self.rank = stages[0].group_rank
         # Set the pipeline stage states
-        if stage_index_to_group_rank is not None:
-            for stage in self._stages:
-                stage.stage_index_to_group_rank = stage_index_to_group_rank
-        self.stage_index_to_group_rank = stages[0].stage_index_to_group_rank
+        if stage_index_to_group_rank is None:
+            self.stage_index_to_group_rank = generate_stage_to_rank_mapping(
+                self.pp_group_size, self._num_stages
```

Comment:
nit: might be worth being explicit that we're requesting "loop" here for readability and for not relying on a defaul